### PR TITLE
timer at top of room page as red decreasing bar

### PR DESCRIPTION
**Pull request by bumblebot1**

### CODE REVIEW

File: src/client/game/progress.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
```
Comment (dwybourn) [inline src/client/game/progress.js]: Should there not be a test(s) for this class?

File: src/client/game/game-reducer.js [hunk 2]
```diff
@@ -18,12 +18,13 @@ const initialState = {
     submission: false,
     roundResults: false,
     finalAnswers: [],
-    resetRound: false
+    resetRound: false,
+	gameParams: {}
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: identation

File: src/client/game/game-reducer.js [hunk 3]
```diff
@@ -134,6 +135,12 @@ const game = (state = initialState, action) => {
                 disableVowel: false
             })
         }
+		case actionTypes.GAME_PARAMETERS:{
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: indetation

File: src/client/game/game-reducer.js [hunk 4]
```diff
@@ -134,6 +135,12 @@ const game = (state = initialState, action) => {
...
+			console.log(action.payload);
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: console.log

File: src/client/game/game-reducer.js (hunk 2 above)
Comment (bumblebot1) [inline src/client/game/game-reducer.js]: i think github messed this up again it does not show up on my pc

Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: This is an editor issue - spaces vs tabs

File: src/server/cloak-server.js [hunk 5]
```diff
@@ -84,6 +84,10 @@ function newMember(user) {
     } else {
         gameNotStartedRefresh(this);
     }
+    user.message('initialGameParams', {
+        answerTime: gameParameters.answerTime,
+    	submitTime: gameParameters.submitTime
```
Comment (kenneth-gray) [inline src/server/cloak-server.js]: indentation here or line above

File: src/client/services/cloak-service.js [hunk 6]
```diff
@@ -132,6 +133,9 @@ function configureAndRun(roomId) {
             },
             resetFinished: () => {
                 dispatch(resetFinished());
+            },
+            initialGameParams: (params) => {
```
Comment (kenneth-gray) [inline src/client/services/cloak-service.js]: Is this being tested?

File: src/client/game/game-reducer.js (hunk 3 above)
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: This file should be tested really - the important thing to test is that the state returned has a different object reference than the one passed in. Maybe hold off on this/ask the PO

File: src/client/game/progress.js (hunk 1 above)
Comment (bumblebot1) [inline src/client/game/progress.js]: I have added tests for this component

File: src/client/services/cloak-service.js (hunk 6 above)
Comment (bumblebot1) [inline src/client/services/cloak-service.js]: To test this do we just need tests for the exported methods or also for the dispatches?

Comment (bumblebot1) [inline src/client/services/cloak-service.js]: This is now tested

**Comment by bumblebot1** [comment]
can we :ship: ?

**Pull request merged**